phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -3521,6 +3524,31 @@ public void upgradeSystemTables(final String url, final Properties props) throws
                     clearCache();
                 }
             }
+
+            try {
+                metaConnection.createStatement().executeUpdate(getTaskDDL());
+            } catch (NewerTableAlreadyExistsException e) {
+
+            } catch (TableAlreadyExistsException e) {
+                long currentServerSideTableTimeStamp = e.getTable().getTimeStamp();
+                if (currentServerSideTableTimeStamp <= MetaDataProtocol.MIN_SYSTEM_TABLE_TIMESTAMP_5_1_0) {
```

Comment:
Keep in mind this change will also likely go into 4.15, so the upgrade will need to happen for either 4.15 or 5.1